File: readability.py
from cs50 import get_string

text = get_string("Text: ")


nw = 1
ns = 0
nl=0
n=len(text)
for i in range(n):
    if text[i] == ' ':
        nw += 1 #counting words

    elif (text[i] >= 'a' and text[i] <= 'z') or ( text[i] >= 'A' and text[i] <= 'Z') :
        nl += 1 #counting letters
        

    elif text[i] == '.' or text[i] == '?' or text[i] == '!':
        ns += 1 #counting sentences

L= (nl/nw)*100

S= (ns /nw)*100

index =((0.0588*L) - (0.296*S)-15.8)
index= round(index)


if  index < 1:
    print("Before Grade 1")

elif index > 16:
    print("Grade 16+")

else:
    print(f"Grade",int(index)")

File: readability.c
#include<stdio.h>
#include<cs50.h>
#include<string.h>
#include<math.h>

int main()
{
    string text= get_string("Text: ");



    int nw = 1, ns = 0,nl=0;


    for( int i=0, n = strlen(text); i < n; i++)
    {
        if (text[i] == 32)
        {
            nw++; //counting words
        }

        else if ((text[i] >= 'a' && text[i] <= 'z') ||( text[i] >= 'A' && text[i] <= 'Z') )
        {
            nl++; //counting letters
        }

        else if (text[i] == '.' || text[i] == '?' || text[i] == '!' )
        {
            ns++; //counting sentences
        }

    }


   float L= 100 * (float) nl / (float) nw;
   
   float S= 100 * (float) ns / (float) nw;

    float index = 0.0588 * (L) - 0.296 * (S) - 15.8;
    index= round(index);

    if ( index < 1)
    {
        printf("Before Grade 1\n");

    }

    else if (index > 16)
    {
        printf("Grade 16+\n");
    }

    else
    {
        printf("Grade %d\n",(int)index);
    }
}